cmd/kubeadm/app/cmd/phases/init/certs.go

```diff
@@ -230,7 +216,21 @@ func runCAPhase(ca *certsphase.KubeadmCert) func(c workflow.RunData) error {
 		if cert, err := pkiutil.TryLoadCertFromDisk(data.CertificateDir(), ca.BaseName); err == nil {
 			certsphase.CheckCertificatePeriodValidity(ca.BaseName, cert)
 
+			// If CA Cert existed while dryrun, copy CA Cert to dryrun dir for later use
+			if data.DryRun() {
+				err := phases.CopyFileTo(filepath.Join(data.CertificateDir(), kubeadmconstants.CACertName), filepath.Join(data.CertificateWriteDir(), kubeadmconstants.CACertName))
+				if err != nil {
+					return err
```

Comment:
we should wrap these errors for some better context:

```suggestion
					return errors.Wrapf(err, "could not copy %s to dry run directory %s, kubeadmconstants.CACertName, data.CertificateWriteDir())
```

Reply:
Yeah, should have thought.
Updated, thanks!